source/grammar/qasm3.g4

```diff
@@ -276,34 +279,73 @@ logicalOperator
     | '||'
     ;
 
+shiftOperator
+    : '<<'
+    | '>>'
+    ;
+
+additiveOperator
+    : '+'
+    | '-'
+    ;
+
+multiplicativeOperator
+    : '*'
+    | '/'
+    | '%'
+    ;
+
+powerOperator
+    : '**'
+    ;
+
 expressionStatement
     : expression SEMICOLON
     ;
 
 expression
     // include terminator/unary as base cases to simplify parsing
     : expressionTerminator
-    | unaryExpression
     // expression hierarchy
-    | xOrExpression
-    | expression '|' xOrExpression
+    | booleanExpression
     ;
 
 /**  Expression hierarchy for non-terminators. Adapted from ANTLR4 C
   *  grammar: https://github.com/antlr/grammars-v4/blob/master/c/C.g4
   * Order (first to last evaluation):
     Terminator (including Parens),
     Unary Op,
+    Power
```

Comment:
Isn't power above unary? Guess that would depend on the convention we choose, but at least in some languages it is